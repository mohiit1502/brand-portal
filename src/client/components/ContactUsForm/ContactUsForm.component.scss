.contact-us-form-row{
  width: 100% !important;
}

