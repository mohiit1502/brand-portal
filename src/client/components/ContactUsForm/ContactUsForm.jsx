import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import './ContactUsForm.component.scss';
import ContentRenderer from "../../utility/ContentRenderer";
import Validator from "../../utility/validationUtil";
import Helper from "../../utility/helper";
import Http from "../../utility/Http";
import {NOTIFICATION_TYPE, showNotification} from "../../actions/notification/notification-actions";

class ContactUsForm extends React.Component{
  constructor(props) {
    super(props);

    const functions = ["onChange","handleSubmit","resetForm","setSelectInputValue"]
    functions.forEach(func => {this[func] = this[func].bind(this)});

    this.validateState = Validator.validateState.bind(this);
    this.fieldRenderer = ContentRenderer.getFieldRenders.bind(this);
    this.onInvalid = Validator.onInvalid.bind(this);
    this.loader = Helper.loader.bind(this);

    this.state = {
      form: {
        inputData: this.props.contactUsForm.fields,
        ...this.props.contactUsForm.formConfig
      }
    };

    this.state.form.inputData.area.tooltipContent = (
      <div className="py-2">
        <b className="position-absolute text-white tooltip-close-button">x</b>
        <p className="mt-2 pl-2 text-left font-size-12">
          <b>Technical Support:</b> Select if you are facing an issue with a system functionality.<br/>
          <br />
          <b>Claim Support:</b> Select if you need help with a specific claim.<br />
          <br/>
          <b>User Management Support:</b> Select if you need help with managing current or new users.<br />
          <br/>
          <b>Follow-Up:</b> Select if you want to follow up on an open ticket.<br />
          <br/>
          <b>IP Management Support:</b> Select if you need help with managing your brands.<br />
        </p>
      </div>);
  }

  componentDidMount() {
    this.resetForm();
  }

  resetForm(){
    const form = {...this.state.form};
    form.inputData.area.value = "";
    form.inputData.title.value = "";
    form.inputData.details.value = "";
    this.setState({form});
  }

  setSelectInputValue (value, key) {
    if (value) {
      this.setState(state => {
        state = {...state};
        state.form.inputData[key].value = value;
        return {
          ...state
        };
      });
    }
  }

  onChange(event,key){
    let targetVal = "";
    if(key === "area"){
      targetVal = event;
    }else if(event && event.target){
      targetVal = event.target.value;
    }
    this.setState(state => {
      state = {...state};
      state.form.inputData[key].value = targetVal;
      return state;
    });

  }

  checkEnableSubmit(){
    const form = {...this.state.form}
    const isSubmitEnabled =  form.inputData.area.value && form.inputData.title.value
            && form.inputData.details.value;
    form.inputData.sendActions.buttons.send.disabled = !isSubmitEnabled;
    this.setState({form});
  }

  handleSubmit(evt){
    evt.preventDefault();
    this.validateState();
    if(!this.validateState()){
      let form = {...this.state.form};
      console.log(this.state.form);
      this.loader("form",true);
      const url = form.api;
      const area = form.inputData.area.value;
      const title = form.inputData.title.value;
      const details = form.inputData.details.value;
      const payload = {area,title,details};
      return Http.post(url,payload).then(res => {
          if(res.body){
            this.resetForm();
            this.props.showNotification(NOTIFICATION_TYPE.SUCCESS,form.successNotificationMessage);
          }else{
            this.props.showNotification(NOTIFICATION_TYPE.ERROR,form.failedNotificationMessage);
          }
          this.loader("form",false);
        }
      ).catch(err => {
        this.loader("form",false);
        this.props.showNotification(NOTIFICATION_TYPE.ERROR,form.failedNotificationMessage);
        console.log(err);
      })
    }
  }

  render(){
    return (
      <form onSubmit={this.handleSubmit} className={`contact-us-form ${this.state.form.loader ? "loader" : ""}`} >
        {this.fieldRenderer()}
      </form>
    )
  };
};

ContactUsForm.propTypes = {
  showNotification: PropTypes.func
};

const mapStateToProps = state => {
  return {
    contactUsForm: state.content && state.content.metadata && state.content.metadata.SECTIONSCONFIG && state.content.metadata.SECTIONSCONFIG.CONTACTUS
  };
}

const mapDispatchToProps = {
  showNotification
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ContactUsForm);
