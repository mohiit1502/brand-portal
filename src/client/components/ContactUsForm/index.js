import ContactUsForm from "./ContactUsForm.jsx";

export default ContactUsForm;
