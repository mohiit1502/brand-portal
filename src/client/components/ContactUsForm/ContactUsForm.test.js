/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import ContactUsForm from "./ContactUsForm";

describe("ContactUsForm", () => {
   it("renders without error", () => {

   });
});
