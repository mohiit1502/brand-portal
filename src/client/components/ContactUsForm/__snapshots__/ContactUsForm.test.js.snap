// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUsForm renders without error 1`] = `
<form
  className="contact-us-form "
  onSubmit={[Function]}
>
  <div
    className="form-row mt-4 contact-us-form-row w-100"
  >
    <div
      className="contact-us-form-area col-5"
    >
      <div
        className="form-group custom-input-form-group custom-select-form-group dropdown  mb-3  field-select-arrow"
      >
        <img
          alt="i"
          className="positioned-top-right"
          data-html="true"
          data-placement="right"
          data-toggle="tooltip"
          src="question.svg"
          title="<div class=\\"py-2\\" data-reactroot=\\"\\"><b class=\\"position-absolute text-white tooltip-close-button\\">x</b><p class=\\"mt-2 pl-2 text-left font-size-12\\"><b>Technical Support:</b> Select if you are facing an issue with a system functionality.<br/><br/><b>Claim Support:</b> Select if you need help with a specific claim.<br/><br/><b>User Management Support:</b> Select if you need help with managing current or new users.<br/><br/><b>Follow-Up:</b> Select if you want to follow up on an open ticket.<br/><br/><b>IP Management Support:</b> Select if you need help with managing your brands.<br/></p></div>"
        />
        <input
          autoComplete="off"
          className="form-control form-control-area custom-input-element"
          data-toggle="dropdown"
          disabled={false}
          id="contactUs-area-custom-input"
          onChange={[Function]}
          required={false}
          type="select"
          value=""
        />
        <label
          className="custom-input-label custom-input-label-placeholder"
          htmlFor="contactUs-area-custom-input"
        >
          <div
            className="label-upper-bg position-absolute w-100 h-50 d-block"
          />
          <div
            className="label-lower-bg position-absolute w-100 h-50 d-block"
          />
          <span
            className="label-text"
          >
             
            Area
             
          </span>
        </label>
        <div
          className="dropdown-menu"
        >
          <a
            className="dropdown-item pt-2"
            onClick={[Function]}
          >
            Technical Support
            <p
              className="dropdown-subtitle m-0"
            />
          </a>
          <a
            className="dropdown-item pt-2"
            onClick={[Function]}
          >
            Claim Support
            <p
              className="dropdown-subtitle m-0"
            />
          </a>
          <a
            className="dropdown-item pt-2"
            onClick={[Function]}
          >
            User Management Support
            <p
              className="dropdown-subtitle m-0"
            />
          </a>
          <a
            className="dropdown-item pt-2"
            onClick={[Function]}
          >
            Follow-Up
            <p
              className="dropdown-subtitle m-0"
            />
          </a>
          <a
            className="dropdown-item pt-2"
            onClick={[Function]}
          >
            IP Management Support
            <p
              className="dropdown-subtitle m-0"
            />
          </a>
          <a
            className="dropdown-item pt-2"
            onClick={[Function]}
          >
            Other
            <p
              className="dropdown-subtitle m-0"
            />
          </a>
        </div>
        <small
          className="form-text custom-input-help-text text-muted"
          style={
            Object {
              "paddingLeft": "0.7rem",
            }
          }
        >
          
        </small>
      </div>
    </div>
  </div>
  <div
    className="form-row  contact-us-form-row w-100"
  >
    <div
      className="contact-us-form-title col-5"
    >
      <div
        className="form-group custom-input-form-group form-group-text mb-3
        "
        style={
          Object {
            "position": "static",
          }
        }
      >
        <input
          className="form-control form-control-title custom-input-element"
          disabled={false}
          id="contactUs-title-custom-input"
          onBlur={[Function]}
          onChange={[Function]}
          onInvalid={[Function]}
          required={false}
          type="text"
          value=""
        />
        
        <label
          className="custom-input-label custom-input-label-placeholder"
          htmlFor="contactUs-title-custom-input"
        >
          <div
            className="label-upper-bg position-absolute w-100 h-50 d-block"
          />
          <div
            className="label-lower-bg position-absolute w-100 h-50 d-block"
          />
          <span
            className="label-text required"
          >
             
            Title
             
          </span>
        </label>
        <small
          className="form-text custom-input-help-text text-muted"
          style={
            Object {
              "paddingLeft": undefined,
            }
          }
        >
          
        </small>
      </div>
    </div>
  </div>
  <div
    className="form-row contact-us-form-row w-100"
  >
    <div
      className="contact-us-form-details col-7"
    >
      <div
        className="form-group custom-input-form-group form-group-textarea "
      >
        <label
          className="custom-input-label custom-input-label-textarea  required"
          htmlFor="contactUs-details-custom-input"
        >
          Details
           
          
        </label>
        <textarea
          className="form-control form-control-details custom-input-element custom-input-element-textarea"
          id="contactUs-details-custom-input"
          onChange={[Function]}
          placeholder="Please provide additional information about the request"
          required={false}
          rows={4}
          value=""
        />
        <small
          className="form-text custom-input-help-text text-area-error text-muted"
          style={
            Object {
              "paddingLeft": undefined,
            }
          }
        >
          
        </small>
      </div>
    </div>
  </div>
  <div
    className="form-row contact-us-form-row w-100 m-0"
  >
    <div
      className="contact-us-button-panel text-right col-7"
    >
      <div
        className="c-ButtonsPanel form-row contact-us-form-row w-100 m-0"
        data-test="button-panel"
      >
        <div
          className="col contact-us-button-panel text-right"
          data-test="button-div"
        >
          <button
            className="btn btn-primary btn-sm px-4 font-size-14"
            type="submit"
          >
            Send
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
